### ISSUE TITLE
Make it pytorch compatible?

### ISSUE THREAD

**Issue by HWiese1980** [rock-learning/pytransform3d#35]
Hi @AlexanderFabisch, do you think it's possible to make `pytransform3d` compatible with `pytorch`? That is, make it aware of gradients and such? 

I would like to use it in a `nn.Module` class, but it seems like it's losing the gradients on the way. Python complains about that `numpy()` cannot be called on tensors that have `requires_grad=True`. It's required to `detach` them before. That seems to happen somewhere in the guts of `pytransform3d`. But as far as I've understood, the tensor must not be detached because it'd get useless for the backward pass. 

In my particular case it's a call to `transform_from_pq`, more precisely, `np.asarray` in `check_pq` that results in `tensor.numpy()` being called.

**Comment by AlexanderFabisch** [comment]
Do you have a simple test script for your use case? Maybe there are more of these issues and I would like to find them. Would be an awesome feature if the library can directly be used with pytorch tensors...

**Comment by HWiese1980** [comment]
Probably. But does it make sense if you use pyTorch everywhere else?

**Comment by AlexanderFabisch** [comment]
This is not in the focus of this package, but maybe an interesting research project. ;)
